hts.c

```diff
@@ -2602,10 +2608,24 @@ hts_itr_t *hts_itr_query(const hts_idx_t *idx, int tid, hts_pos_t beg, hts_pos_t
             } while (bin);
             if (bin == 0) k = kh_get(bin, bidx, bin);
             min_off = k != kh_end(bidx)? kh_val(bidx, k).loff : 0;
+            // min_off can be calculated more accurately if the
+            // linear index is available
             if (idx->lidx[tid].offset
-                && beg>>idx->min_shift < idx->lidx[tid].n
-                && min_off < idx->lidx[tid].offset[beg>>idx->min_shift])
-                min_off = idx->lidx[tid].offset[beg>>idx->min_shift];
+                && rel_off < idx->lidx[tid].n
+                && min_off < idx->lidx[tid].offset[rel_off])
+                min_off = idx->lidx[tid].offset[rel_off];
+            if (unmapped) {
```

Comment:
I think this should be contained within the if statement above, and the line above should be an else clause - mainly because it uses `idx->lidx[tid].offset` and I think that doesn't exist in CSI?

Reply:
Agree with the nesting of `if(unmapped)` inside `if (idx->lidx[tid].offset && rel_off < idx->lidx[tid].n)`.
I am not sure about the `else`, though. I need `min_off = idx->lidx[tid].offset[rel_off]` to be the staring point for searching a new value.

Reply:
Ah, yes.  I see it does use it as a starting point - in which case the loop should probably start at `rel_off - 1`.